### PR TITLE
feat: integrate STAC-Catalog feature

### PR DESCRIPTION
**Pull request by RanbirAulakh**
**Issue #, if available:** n/a

### Notes
- Integrate STAC-Catalog feature

### Testing
- Check `Checks` tab since this [PR](https://github.com/aws-solutions-library-samples/osml-cdk-constructs/pull/114) has been merged in

Before you submit a pull request, please make sure you have to following:

- [x] Your code contains tests that cover all new code and changes
- [x] All new and existing tests passed
- [x] I have read the [Contributing Guidelines](https://github.com/aws-solutions-library-samples/guidance-for-overhead-imagery-inference-on-aws/blob/main/CONTRIBUTING.md) and agree to follow the [Code of Conduct](
https://github.com/aws-solutions-library-samples/guidance-for-overhead-imagery-inference-on-aws/blob/main/CODE_OF_CONDUCT.md))

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/osml-stacks/stac-catalog/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+RUN dnf upgrade --refresh -y && \
+    dnf install gcc -y && \
+    dnf clean all
+
+ENV CURL_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
+
+WORKDIR /app
+
+COPY .. /app
```
Comment (drduhe) [inline lib/osml-stacks/stac-catalog/Dockerfile:13]: Remove this line

**Review by drduhe — CHANGES_REQUESTED** [review]

File: bin/deploy-data-catalog.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      account: targetAccount,
+      osmlVpc: vpcStack.resources,
+      dockerImageCode: containerStack.resources.dockerImageCode,
+      description:
+        "Data Catalog Dataplane, Guidance for Overhead Imagery Inference on AWS (SO9240)"
+    }
+  );
+
+  dataplaneStack.addDependency(containerStack);
+}
```
Comment (drduhe) [inline bin/deploy-data-catalog.ts:61]: Blank line needed

**Review by drduhe — CHANGES_REQUESTED** [review]

File: bin/deploy-data-catalog.ts (hunk 2 above)
Comment (RanbirAulakh) [reply bin/deploy-data-catalog.ts]: fixed in latest revision.

**Review by drduhe — APPROVED** [review]
LGTM

**Pull request merged**